core/controllers/topic_editor.py

```diff
@@ -149,8 +151,45 @@ def get(self, topic_id: str) -> None:
         additional_story_summary_dicts = [
             summary.to_dict() for summary in additional_story_summaries]
 
+        canonical_stories_ids = [summary['id'] for
+            summary in canonical_story_summary_dicts]
+        canonical_stories = story_fetchers.get_stories_by_ids(
+            canonical_stories_ids)
+        non_empty_canonical_stories = [story for story in
+            canonical_stories if story is not None]
         updated_canonical_story_summary_dicts = []
         for summary in canonical_story_summary_dicts:
+            story = None
+            for canonical_story in non_empty_canonical_stories:
+                if canonical_story.id == summary['id']:
+                    story = canonical_story
+            if story is None:
+                continue
```

Comment:
What is the maximum expected length of `canonical_story_summary_dicts` and `non_empty_canonical_stories`? Would it make more sense to create a dictionary of `{canonical_story.id: canonical_story}` and extract story like this:
```python
story = canonical_story_dict.get(summary['id'])
if story is None:
    continue
```

Reply:
Typically the max length is pretty small, maybe 0-2 for a given topic. We expect at most 15 topics or so per classroom.

Reply:
Done